pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -491,6 +492,16 @@ func (c *Controller) preparePacket(tf *crdv1beta1.Traceflow, intf *interfacestor
 		} else if packet.DestinationIP.To4() != nil {
 			return nil, errors.New("destination Service does not have an IPv6 ClusterIP")
 		}
+		if !liveTraffic {
+			switch dstSvc.Spec.Ports[0].Protocol {
+			case corev1.ProtocolTCP:
+				packet.IPProto = protocol.Type_TCP
+				packet.TCPFlags = uint8(2)
+			case corev1.ProtocolUDP:
+				packet.IPProto = protocol.Type_UDP
```

Comment:
The switch statement does not include a default case. While the current implementation handles TCP and UDP, if the service protocol is SCTP which is also currently not supported , we could add a default case !
 default:
	// Handle unsupported protocols, maybe log a warning

Reply:
defaults to ICMP which is consistent with other cases also.
In this case, the behaviour will be run traceflow as Pod-to-IP(Service's ClusterIP) with ICMP protocol.